### PR TITLE
Subscription Restored goes through handshaking

### PR DESCRIPTION
**Pull request by kleewho**

### CODE REVIEW

File: src/main/kotlin/com/pubnub/api/enums/PNStatusCategory.kt [hunk 1]
```diff
@@ -94,4 +94,9 @@ enum class PNStatusCategory {
      * The subscription has been stopped.
      */
     PNDisconnectedCategory,
+
+    /**
+     * Previously started subscribe loop did fail and at this moment client disconnected from real-time data channels.
```
Comment (techwritermat) [inline src/main/kotlin/com/pubnub/api/enums/PNStatusCategory.kt:99]:
```suggestion
     * Previously started subscribe loop failed, and at this moment client disconnected from real-time data channels.
```

**Review by marcin-cebo — APPROVED** [review]

**Review by marcin-cebo — APPROVED** [review]

File: src/test/kotlin/com/pubnub/api/subscribe/eventengine/worker/TransitionFromHandshakeFailedStateTest.kt [hunk 2]
```diff
@@ -60,10 +60,10 @@ class TransitionFromHandshakeFailedStateTest {
 
         // then
         assertEquals(
-            SubscribeState.Receiving(channels, channelGroups, subscriptionCursor), state
+            SubscribeState.Handshaking(channels, channelGroups, subscriptionCursor), state
```
Comment (marcin-cebo) [inline src/test/kotlin/com/pubnub/api/subscribe/eventengine/worker/TransitionFromHandshakeFailedStateTest.kt:63]: test name probably should be changed as well to:
can_transit_from_HANDSHAKE_FAILED_to_HANDSHING_when_there_is_SUBSCRIPTION_RESTORED_even

File: src/test/kotlin/com/pubnub/api/subscribe/eventengine/worker/TransitionFromHandshakingReconnectingStateTest.kt [hunk 3]
```diff
@@ -140,11 +140,11 @@ class TransitionFromHandshakingReconnectingStateTest {
         )
 
         // then
-        assertEquals(SubscribeState.Receiving(channels, channelGroups, subscriptionCursor), state)
+        assertEquals(SubscribeState.Handshaking(channels, channelGroups, subscriptionCursor), state)
```
Comment (marcin-cebo) [inline src/test/kotlin/com/pubnub/api/subscribe/eventengine/worker/TransitionFromHandshakingReconnectingStateTest.kt:143]: Test name should be renamed to reflect this change?

File: src/test/kotlin/com/pubnub/api/subscribe/eventengine/worker/TransitionFromHandshakingStateTest.kt [hunk 4]
```diff
@@ -62,11 +62,11 @@ class TransitionFromHandshakingStateTest {
         )
 
         // then
-        assertEquals(SubscribeState.Receiving(channels, channelGroups, subscriptionCursor), state)
+        assertEquals(SubscribeState.Handshaking(channels, channelGroups, subscriptionCursor), state)
```
Comment (marcin-cebo) [inline src/test/kotlin/com/pubnub/api/subscribe/eventengine/worker/TransitionFromHandshakingStateTest.kt:65]: Test name should be renamed to reflect change?

File: src/test/kotlin/com/pubnub/api/subscribe/eventengine/worker/TransitionFromUnsubscribedStateTest.kt [hunk 5]
```diff
@@ -40,10 +40,10 @@ class TransitionFromUnsubscribedStateTest {
         )
 
         // then
-        assertEquals(SubscribeState.Receiving(channels, channelGroups, subscriptionCursor), state)
+        assertEquals(SubscribeState.Handshaking(channels, channelGroups, subscriptionCursor), state)
```
Comment (marcin-cebo) [inline src/test/kotlin/com/pubnub/api/subscribe/eventengine/worker/TransitionFromUnsubscribedStateTest.kt:43]: Test name should be renamed to reflect change?

**Pull request merged**